src/deploy/osp_deployer/powerflexgw.py

```diff
@@ -0,0 +1,250 @@
+                    "ssh " + ssh_opts + " heat-admin@" +
+                    ip + " sudo cat " + self.lia_cert +
+                    " | sshpass -p " + self.root_pwd +
+                    " ssh root@" + self.ip +
+                    ' " cat > ' + self.certs_dir + "/" + hostname + '.lia.cer"']
+                for cmd in cmds:
+                    re = self.director.run(cmd)
+
+            else:
+                logger.debug("Capturing LIA certificate on host {}".format(hostname))
+                cmd = ("ssh " + ssh_opts + " heat-admin@" +
+                       ip + 
+                       " sudo cat " + self.lia_cert +
+                       " | sshpass -p " + self.root_pwd +
+                       " ssh root@" + self.ip +
+                       ' " cat > ' + self.certs_dir + "/" + hostname + '.lia.cer"')
+                re = self.director.run(cmd)
+
+ 
+    def inject_ssl_certificates(self):
+    
+        certs_file_count = 0
+        
+        cmd = (" ls " + self.certs_dir)
+        re = self.run_as_root(cmd)
+        certs = re[0].split("\n") 
+        certs.pop()
+ 
+        for cert_file in certs:
+            certs_file_count += 1
+            alias = cert_file.rsplit(".", 1)[0]
+            cmds = [
+               "sed -ni '/CERTIFICATE/,/CERTIFICATE/p' " + 
+               self.certs_dir + "/" + cert_file,
+               "keytool -import -trustcacerts -noprompt -storepass " +
+               self.keystore_pwd + " -alias " +
+               alias + " -file " + self.certs_dir + "/" + cert_file +
+               " -keystore " + self.keystore]
+            logger.debug("Importing certificate for host {} as alias {}".format(alias.split(".")[0], alias))
+            for cmd in cmds:
+               re = self.run_as_root(cmd) 
+
+        cmd = ("keytool -noprompt -storepass " +
+               self.keystore_pwd + " -list -keystore " +
+               self.keystore + " | grep trusted")
+        re = self.run_as_root(cmd)
+        certs_count = re[0].split("\n") 
+        certs_count.pop()
+        if len(certs_count) == certs_file_count:
+            logger.debug("All certificates have been successfully imported")
+
+
+    def restart_gateway(self):
+      
+        logger.debug("Restarting gateway service")
+        cmd = ("systemctl restart scaleio-gateway")
+
+        self.run_as_root(cmd)
+
+
+    def restart_cinder_volume(self):
+
+        ssh_opts = (
+        " -o StrictHostKeyChecking=no "
+        " -o UserKnownHostsFile=/dev/null "
+        " -o KbdInteractiveDevices=no"
+        " -o LogLevel=ERROR")
+
+        logger.debug("Restarting cinder-volume service")
+        cmd = ("ssh " + ssh_opts + " heat-admin@" +
+               self.controller_vip + " sudo pcs resource restart openstack-cinder-volume")
+        self.director.run(cmd)
+
+        re = self.director.run(self.source_overcloudrc +
+                          "openstack volume service list | grep powerflex")
```

Comment:
The volume service name is the backend name specified in the THT powerflex, user can change it. Cannot assume it to be powerflex

Reply:
Part of a new commit. Value is now based up settings. Fixed.

Reply:
Thanks